Lesson 4/src/app/components/WelcomeModal.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {faTimes} from '@fortawesome/free-solid-svg-icons';
+
+export default class WelcomeModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            modal: false
+        };
+        console.log('1. constructor()');
+    }
+
+    static getDerivedStateFromProps(props, state){
+        console.log('2. getDerivedStateFromProps(props, state)');
+        return null
+    }
+
+    render() {
+        console.log('3. render()');
+        let modal;
+        if (this.state.modal){
+            modal =
+                <>
+                    <div className={'WelcomeModal'}>
```

Comment:
```suggestion
                    <div className="WelcomeModal">
```
зачем задавать значение класса с помощью JS-конструкции, если можно просто указать текст:)